test/integration/nss-tests.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+set -xo pipefail
+
+# SPDX-License-Identifier: BSD-2-Clause
+CA_PEM="$TEST_FIXTURES/ca.pem"
+CA_KEY="$TEST_FIXTURES/ca.key"
+
+CLIENT_CNF="$TEST_FIXTURES/client.cnf"
+PASSWORD_CA=whatever
+EXT_FILE="$TEST_FIXTURES/smimeextensions"
+
+export NSS_DEFAULT_DB_TYPE=sql
+
+if ! command -v certutil ||
+   ! command -v modutil; then
+  # Skip this test unless certutil/modutil are found
+  exit 77
+fi
+
+function pinentry() {
+  pin="$1"
+  shift
+  cmd="$*"
+  printf 'spawn %s\nexpect "Password or Pin"\nsend -- %s\\r\nexpect eof\n' \
+	  "$cmd" "$pin" | expect
+}
+
+function cleanup() {
+  if [ "$1" != "no-kill" ]; then
+      pkill -P $$ || true
+  fi
+  rm -f index.txt index.txt.attr serial serial.old index.txt.old index.txt.attr.old \
+        03.pem smimeclient.csr smimeclient.crt smimeclient.key smimeclient.pem \
+	pkcs11.txt cert9.db key4.db userpin.txt
+}
+trap cleanup EXIT
+
+onerror() {
+  echo "$BASH_COMMAND on line ${BASH_LINENO[0]} failed: $?"
+  exit 1
+}
+trap onerror ERR
+
+cleanup "no-kill"
+
+if [ -z "$modpath" ]; then
+  modpath="$PWD/src/.libs/libtpm2_pkcs11.so"
+fi
+
+echo "modpath=$modpath"
+
+# make the DB
+touch index.txt
+touch index.txt.attr
+echo "03" >> serial
```

Comment:
copy paste error? Why are you creating the db? Does something below use it?

Reply:
It's required by `openssl ca` below. Usually, `tls-tests.sh` (where this part and much of the framework as a whole is copied from) will have it set up already, but this code makes sure `nss-tests.sh` can also be called stand-alone.

Reply:
Ahh I scanned for that command, I missed it.